src/pkgcheck/checks/metadata.py

```diff
@@ -747,7 +747,7 @@ def feed(self, pkg):
                     in_or_restriction = False
 
                 for atom in iflatten_instance(node, (atom_cls,)):
-                    if self.deprecated(atom):
+                    if not atom.blocks and self.deprecated(atom):
```

Comment:
You might want to add a comment here explaining 'why'- it took me a bit to think through why you were adding this.  I agree with it- at first glance- but a comment would've made the understanding quicker, including for folks who look at this down the line.

Reply:
Done.